biogrid/libs/biogrid-simulator/src/lib/biogrid-action/biogrid-action-class.ts

```diff
@@ -2,10 +2,15 @@ import { GridAction, SupplyingPath } from '@biogrid/grid-simulator';
 
 export class BiogridAction implements GridAction {
   private supplyPath: SupplyingPath;
-  constructor(supplyPath: SupplyingPath) {
+  private efficiency: number;
+  constructor(supplyPath: SupplyingPath, efficiency: number) {
     this.supplyPath = supplyPath;
+    this.efficiency = efficiency;
   }
```

Comment:
According to our internal TS style guide (which I unfortunately cannot link you to), you can simplify this since you're doing a 1:1 conversion of the constructor params to class properties:

```
export class BiogridAction implements GridAction {
  constructor(
    private supplyPath: SupplyingPath,
    private efficiency: number
  ) {}

  ...
}
```

The style guide implies that this also sets the class properties automatically, although I haven't personally tried this out

Reply:
See https://www.typescriptlang.org/docs/handbook/classes.html#parameter-properties for a reference that you can access